Social: Move Share post action from post-list to Publicize

gmjuhasz opened the pull request:
Fixes SOCIAL-16

## Proposed changes:

* Move the Share post action from the `post-list` package to the `Publicize` package for better discoverability
* The Share action now appears automatically for users with plans that support the `republicize` feature
* Maintain backward compatibility with the existing `jetpack_post_list_display_share_action` filter

### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

1. On a site with a plan that supports `republicize` (Jetpack Social or higher):
   - Go to Posts > All Posts
   - Hover over a published post
   - Verify the "Share" action appears in the row actions
   - Click Share and verify it opens the editor with the share panel

2. On a site without `republicize` support:
   - Verify the Share action does NOT appear by default
   - Add `add_filter( 'jetpack_post_list_display_share_action', '__return_true' );` to enable it
   - Verify the Share action now appears (backward compatibility)

3. Verify the Share action only appears for:
   - Published posts (not drafts)
   - Post types that support `publicize`
   - Post types that use the block editor
   - Social Notes

gmjuhasz committed:
Social: Move Share post action from post-list to Publicize

Move the Share post action functionality from the post-list package
to the Publicize package for better discoverability. The Share action
now appears automatically for users with plans that support the
'republicize' feature, while maintaining backward compatibility with
the existing `jetpack_post_list_display_share_action` filter.

gmjuhasz committed:
Merge branch 'trunk' into SOCIAL-16

manzoorwanijk reviewed (commented):
This PR is a good and timely reminder that we need to wire up this with the unified modal.

manzoorwanijk commented:
It's done here
https://github.com/Automattic/jetpack/blob/5988af7f799aab67b500fd596b2140cb598ab7cc/projects/js-packages/publicize-components/src/components/block-editor/shared-utils.ts#L24-L25

gmjuhasz committed:
Merge branch 'trunk' into SOCIAL-16

manzoorwanijk commented on projects/packages/publicize/src/class-publicize-setup.php:
```diff
@@ -183,9 +183,63 @@ public static function add_filters_and_actions_for_screen( $current_screen ) {
+		 * @param string $post_type  The current post type.
+		 */
+		$show_share_action = Current_Plan::supports( 'republicize' )
+			|| apply_filters( 'jetpack_post_list_display_share_action', false, $current_screen->post_type );
```
Previously, if I as a developer wanted to remove that "Share" action, I could do this

```php
add_filter( 'jetpack_post_list_display_share_action', '__return_false', 20 );
```

Now it doesn't work because of the OR condition. IMHO, it should be


```suggestion
		$show_share_action = Current_Plan::supports( 'republicize' )
			&& apply_filters( 'jetpack_post_list_display_share_action', true, $current_screen->post_type );
```

gmjuhasz replied:
Yeah good catch. Let's do that to retain that functionality, maybe someone uses it

gmjuhasz replied:
Fixed in c3cdab7

manzoorwanijk reviewed (commented):
This works well but I have a suggestion about a minor improvement to retain the current behavior.

Also, there is some logic related to it (`jetpack_post_list_display_share_action`) in the Social plugin as well. Should we clean that up?

gmjuhasz committed:
Fix Share_Action_Test to clean up Current_Plan cache

Add proper tear_down() method that clears the Current_Plan static
cache using reflection to prevent test isolation issues with
subsequent tests that depend on plan feature checks.

gmjuhasz committed:
Change Share action filter logic from OR to AND

gmjuhasz committed:
Update Share action tests for new filter logic

gmjuhasz commented on projects/packages/publicize/src/class-publicize-setup.php:
```diff
@@ -183,9 +183,63 @@ public static function add_filters_and_actions_for_screen( $current_screen ) {
+			use_block_editor_for_post_type( $post_type )
+		) {
+			add_filter( 'post_row_actions', array( self::class, 'add_share_action' ), 15, 2 );
+			add_filter( 'page_row_actions', array( self::class, 'add_share_action' ), 15, 2 );
```
Note for the future: Dont trust AI for moving filters over, because it changes things :D

gmjuhasz replied:
Fixed in c3cdab7

gmjuhasz commented on projects/packages/publicize/tests/php/Share_Action_Test.php:
```diff
@@ -0,0 +1,132 @@
+		Publicize_Setup::add_filters_and_actions_for_screen( $current_screen );
+
+		$this->assertFalse( has_action( 'post_row_actions', array( Publicize_Setup::class, 'add_share_action' ) ) );
+	}
```
Added test for it in d01faf5

gmjuhasz committed:
Social: Remove redundant share action filter hook

gmjuhasz committed:
Wpcomsh: Remove redundant share action filter hook

manzoorwanijk reviewed (dismissed):
Works fine on Jetpack, Atomic, and Simple ✅ 

Thank you for the changes.

manzoorwanijk commented on projects/plugins/wpcomsh/feature-plugins/post-list.php:
```diff
@@ -16,34 +16,10 @@
  */
 function wpcomsh_post_list_init() {
 	add_filter( 'jetpack_block_editor_republicize_feature', '__return_true' );
-	wpcomsh_maybe_enable_share_action();
```
Will this change cause issues on Atomic when they receive the wpcomsh change today but Jetpack change on Monday?

manzoorwanijk replied:
Yes, my apprehension is correct. I tested this change with wpcomsh from this branch and Jetpack RC and the Share button is gone.

<img width="472" height="97" alt="image" src="https://github.com/user-attachments/assets/8c9fddab-4cf6-4c83-8c70-d656179f75b4" />

So, it means that we may need to hold off merging this till Monday morning.

manzoorwanijk replied:
Otherwise, we can extract the wpcomsh change into a separate PR and merge that after the next Atomic deployment. That will unblock this PR.

gmjuhasz replied:
I think it's fine to wait until Monday, it's not urgent to ship, and not likely to get conflicts either

manzoorwanijk replied:
Sounds good. Thanks

gmjuhasz committed:
Remove deprecated setAccessible call for PHP 8.5

gmjuhasz committed:
Fix PHP version compatibility for setAccessible call

grzegorz-cp commented:
Looks good, nice that you noticed the issue @manzoorwanijk 👍
